internal/onepassword/model/item.go

```diff
@@ -0,0 +1,291 @@
+package model
+
+import (
+	"context"
+	"fmt"
+
+	connect "github.com/1Password/connect-sdk-go/onepassword"
+	sdk "github.com/1password/onepassword-sdk-go"
+)
+
+type Item struct {
+	ID       string
+	Title    string
+	VaultID  string
+	Category connect.ItemCategory
+	Version  int
+	Tags     []string
+	URLs     []ItemURL
+	Sections []ItemSection
+	Fields   []ItemField
+	Files    []ItemFile
+}
+
+type ItemSection struct {
+	ID    string
+	Label string
+}
+
+type ItemField struct {
+	ID       string
+	Label    string
+	Type     connect.ItemFieldType
+	Value    string
+	Purpose  connect.ItemFieldPurpose
+	Section  *ItemSection
+	Recipe   *GeneratorRecipe
+	Generate bool
+}
+
+type GeneratorRecipe struct {
+	Length        int
+	CharacterSets []string
+}
+
+type ItemURL struct {
+	URL     string
+	Label   string
+	Primary bool
+}
+
+// FromSDKItemToModel creates a new Item from an SDK item
+func (i *Item) FromSDKItemToModel(item *sdk.Item) {
+	if item == nil {
+		return
+	}
+
+	i.ID = item.ID
+	i.Title = item.Title
+	i.VaultID = item.VaultID
+	i.Category = connect.ItemCategory(item.Category)
+	i.Tags = item.Tags
+	i.URLs = fromSDKURLs(item.Websites)
+
+	// Convert sections/fields/files
+	sectionMap := make(map[string]ItemSection)
+	i.Files = fromSDKFiles(item, sectionMap)
+	i.Sections = fromSDKSections(item, sectionMap)
+	i.Fields = fromSDKFields(item, sectionMap)
```

Comment:
Seems that `sectionMap` is populated only in `i.Sections = fromSDKSections(item, sectionMap)`.

That means when it executes `i.Files = fromSDKFiles(item, sectionMap)` line `sectionMap` will always be empty.

Do we want to re-order them? Or explicitly populate a map with values, so it's easy to catch such cases?

I'd go with option 2.